### DISCUSSION TITLE
Question DS

### DISCUSSION THREAD

**Discussion by Pyroshiv0** [mp2i-fsm/mp2i-2021#139] [Question] [answered]
bonjour monsieur,j'ai essayé de réecrire des fonctions du DS et j'ai testé ça:

``int c_to_i(char c){return ("%d",c-97);}``

le but du programme est de convertir un char en entier
Je me demandais un truc
si on peut utiliser le %d en dehors d'un print,quelle est la fonction qu'on peut lui attribuer(niveau mémoire) et le rôle des autres %

**Comment by fortierq** [comment, marked as answer]
Bonjour,

J'ai mis en ligne une proposition de corrigé : https://mp2i-fsm.github.io/mp2i-2021/ds/ds4/ds4
Il n'est pas possible d'utiliser "%d" en dehors d'un printf, mais return c - 97 convient.
